src/order/boolean_algebra.lean

```diff
@@ -140,6 +140,18 @@ lemma sdiff_le : x \ y ≤ x :=
 calc x \ y ≤ (x ⊓ y) ⊔ (x \ y) : le_sup_right
        ... = x                 : sup_inf_sdiff x y
 
+lemma sdiff_lt (hx : y ≤ x) (hy : y ≠ ⊥) :
+  x \ y < x :=
+begin
+  refine sdiff_le.lt_of_ne (λ h, hy _),
+  suffices hyx : disjoint y x,
+  { rw [disjoint.comm, disjoint_iff] at hyx,
+    rw [eq_comm, ←hyx, inf_eq_right],
+    exact hx },
+  convert (disjoint_inf_sdiff : disjoint (x ⊓ y) _),
+  exacts [(inf_eq_right.2 hx).symm, h.symm],
+end
```

Comment:
This proof is much shorter if you put it on line 340, as we have the lemma that corresponds to the `suffices`:
```lean
lemma sdiff_lt' (hx : y ≤ x) (hy : y ≠ ⊥) :
  x \ y < x :=
begin
  refine sdiff_le.lt_of_ne (λ h, hy _),
  rw [sdiff_eq_self_iff_disjoint', disjoint_iff] at h,
  rw [←h, inf_eq_right.mpr hx],
end
```